### PR TITLE
feat(content-docs): allow SEO metadata for category index pages

### LINKED ISSUE #6238: SEO Metadata for generated-index pages
### Have you read the Contributing Guidelines on issues?

- [X] I have read the [Contributing Guidelines on issues](https://github.com/facebook/docusaurus/blob/main/CONTRIBUTING.md#reporting-new-issues).

### Description

I want og:image for category index pages! docusaurus can also add support for different head tags if possible.

### Has this been requested on Canny?

no

### Motivation

This feature can help in SEO a lot. It's not requested by a lot of users, but it will be very useful like in adding different head tags on generated-index pages, for eg one can use og:image, share links to social apps and can gather traffic to their docusaurus website.

### API design

_No response_

### Have you tried building it?

No I haven't tried and I can't, I am beginner just learning basics.

### Self-service

- [ ] I'd be willing to contribute this feature to Docusaurus myself.

### PR DESCRIPTION
**Pull request by Josh-Cena**
## Motivation

Close #6238.

### Have you read the [Contributing Guidelines on pull requests](https://github.com/facebook/docusaurus/blob/main/CONTRIBUTING.md#pull-requests)?

Yes

## Test Plan

Dogfooding

## TODO

- [ ] Can't figure out how to turn the image into a `require` call: `require('/img/docusaurus.png')` doesn't seem to work despite we have `resolve.roots`?
- [ ] Documentation

### CODE REVIEW

File: packages/docusaurus-plugin-content-docs/src/categoryGeneratedIndex.ts [hunk 1]
```diff
@@ -28,6 +28,8 @@ function getCategoryGeneratedIndexMetadata({
   return {
     title: category.link.title ?? category.label,
     description: category.link.description,
+    image: category.link.image,
+    keywords: category.link.keywords,
```
[the remark this answers is not in the document]
Comment (Josh-Cena) [reply packages/docusaurus-plugin-content-docs/src/categoryGeneratedIndex.ts:32]: These are the two that the SEO component actually accepts

**Comment by slorber** [comment]
ok, that seems good enough 👍 thanks

**Comment by pranay01** [comment]
are the docs updated with how to use this feature? Can it be used to set metadata for blog folder index page?

**Comment by slorber** [comment]
@pranay01 this feature is related to docs category, it is not related in any way to the blog

**Comment by slorber** [comment]
Doc was missing so added in https://github.com/facebook/docusaurus/pull/6708

**Pull request merged**